packages/react-router-dom/__tests__/data-browser-router-test.tsx

```diff
@@ -3000,6 +3000,280 @@ function testDomRouter(
         });
       });
 
+      describe("submitting to self from parent/index when ?index param exists", () => {
+        it("useSubmit", async () => {
+          let router = createTestRouter(
+            createRoutesFromElements(
+              <Route
+                id="parent"
+                path="/parent"
+                element={<Parent />}
+                action={({ request }) => "PARENT ACTION: " + request.url}
+              >
+                <Route
+                  id="index"
+                  index
+                  element={<Index />}
+                  action={({ request }) => "INDEX ACTION: " + request.url}
+                />
+              </Route>
+            ),
+            {
+              window: getWindow("/parent?index&index=keep"),
```

Comment:
When we have an `?index` around from a prior submission, we were incorrectly letting that apply to a `submit(data, { method: 'post' })` in the parent component and then it would try to submit to the child index route action and not the proper contextual parent route action

Reply:
This was handled properly in `useFormAction` (so `Form`/`fetcher.Form` didn't have the bug) but when using `useSubmit` it goes straight to the router which does the path resolving, so we were missing the logic in `router.navigate`/`router.fetch`